.travis.yml

```
@@ -0,0 +1,60 @@
+language: go
+
+go:
+  - '1.12.x'
+env:
+  global:
+    - GO111MODULE=on
+stages:
+  - name: test
+  - name: release-staging
+    if: branch = staging AND type = push
+  - name: deploy-staging
+    if: branch = staging AND type = push
+  - name: release-production
+    if: tag IS present
+  - name: deploy-production
+    if: tag IS present
+
+go_import_path: github.com/src-d/landing
+
+jobs:
+  include:
+    - name: 'Test'
+      stage: test
+      env:
+        - LANDING_URL="//sourced.tech"
+      script:
+        - make build
+        - make packages
+
+    - name: 'Push image to Docker Hub'
+      stage: release-staging
+      env:
+        - LANDING_URL="//landing-staging.srcd.run"
+      script:
+        - make build
+        - make packages
+        - make docker-push
+    - name: 'Push production image to Docker Hub'
+      stage: release-production
+      env:
+        - LANDING_URL="//sourced.tech"
+      script:
+        - make build
+        - make packages
+        - make docker-push
+
+    - name: 'Deploy to staging'
+      stage: deploy-staging
+      install:
+        - make install-helm
+      script:
+        - HELM_RELEASE=landing HELM_CHART=landing HELM_ARGS="--repo https://src-d.github.io/charts/infra/ --set image.tag=dev-${TRAVIS_COMMIT:0:7} -f .helm_staging.yml" make deploy
+    
+    - name: 'Deploy to production'
+      stage: deploy-production
+      install:
+        - make install-helm
+      script:
+        - CLUSTER_ENDPOINT=$PRODUCTION_CLUSTER_ENDPOINT B64_CA_CRT=$PRODUCTION_B64_CA_CRT SERVICE_ACCOUNT_TOKEN=$PRODUCTION_SERVICE_ACCOUNT_TOKEN HELM_RELEASE=landing HELM_CHART=landing HELM_ARGS="--repo https://src-d.github.io/charts/infra/ --set image.tag=$TRAVIS_TAG -f .helm_staging.yml" make deploy
```

```suggestion
        - CLUSTER_ENDPOINT=$PRODUCTION_CLUSTER_ENDPOINT B64_CA_CRT=$PRODUCTION_B64_CA_CRT SERVICE_ACCOUNT_TOKEN=$PRODUCTION_SERVICE_ACCOUNT_TOKEN HELM_RELEASE=landing HELM_CHART=landing HELM_ARGS="--repo https://src-d.github.io/charts/infra/ --set image.tag=$TRAVIS_TAG -f .helm_production.yml" make deploy
```

---

imo it would be easier to compare with `staging` rule if this one would be reordered using the `staging` order:
```suggestion
        - HELM_RELEASE=landing HELM_CHART=landing HELM_ARGS="--repo https://src-d.github.io/charts/infra/ --set image.tag=$TRAVIS_TAG -f .helm_production.yml" CLUSTER_ENDPOINT=$PRODUCTION_CLUSTER_ENDPOINT B64_CA_CRT=$PRODUCTION_B64_CA_CRT SERVICE_ACCOUNT_TOKEN=$PRODUCTION_SERVICE_ACCOUNT_TOKEN make deploy
```

---

So both combined
```suggestion
        - HELM_RELEASE=landing HELM_CHART=landing HELM_ARGS="--repo https://src-d.github.io/charts/infra/ --set image.tag=$TRAVIS_TAG -f .helm_staging.yml" CLUSTER_ENDPOINT=$PRODUCTION_CLUSTER_ENDPOINT B64_CA_CRT=$PRODUCTION_B64_CA_CRT SERVICE_ACCOUNT_TOKEN=$PRODUCTION_SERVICE_ACCOUNT_TOKEN make deploy
```